test/middleware/authorization_middleware/authorization_middleware_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:logotime/middleware/authorization_middleware.dart';
+import 'package:logotime/network/network_result.dart';
+import 'package:logotime/network/user/user_network_service.dart';
+import 'package:logotime/redux/action/common/operation_result_action.dart';
+import 'package:logotime/redux/action/navigation_action.dart';
+import 'package:logotime/redux/action/user/authorization_action.dart';
+import 'package:logotime/redux/state/app/app_state.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:redux/redux.dart';
+import '../../mock/next_dispatcher_mock.dart' as mock;
+import 'authorization_middleware_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<IUserNetworkService>(),
+  MockSpec<Store<AppState>>(),
+])
+void main() {
+  late AuthorizationMiddleware sut;
+  late NextDispatcher next;
+  late MockIUserNetworkService userNetworkService;
+  late Store<AppState> store;
+
+  setUp(() {
+    userNetworkService = MockIUserNetworkService();
+    next = mock.NextDispatcherMock();
+    store = MockStore();
+    sut = AuthorizationMiddleware(userNetworkService: userNetworkService);
+  });
+
+  test('Authorization middleware - log in success', () async {
+    //Arrange
+    when(
+      userNetworkService.logIn(
+        email: anyNamed('email'),
+        password: anyNamed('password'),
+      ),
+    ).thenAnswer(
+      (_) async => Success({}),
+    );
+    when(store.state).thenReturn(AppState.initial().rebuild(
+      (state) => state.authorizationState
+        ..email = 'test@test.com'
+        ..password = 'Qwerty123',
+    ));
```

Comment:
you can move this into `setUp()`. Why do you rebuild state, isn't `AppState.initial()` is not enough?

Reply:
I am rebuilding state, because the email and password are unwraped with `!`, as they are passed to the method after they have passed the validation (meaning they are not null)

Reply:
I changed parameter types to String?, as it won't affect the functionality and removed the rebuild part